### ISSUE TITLE
Search in closed tabs, history and bookmarks

### ISSUE THREAD

**Issue by mahdix** [janraasch/tab-ahead#91]

**Comment by janraasch** [comment]
The request for bookmarks came up once before in https://github.com/janraasch/tab-ahead/issues/35. What I wrote in my first reply there still holds today, however there has to been any more action on the issue which is why I closed it.

Technically a chrome extension can provide the user with anything that can be accessed via the `chrome`-API. I would need to have a look at the docs, but I think we should be able to get permission for history and bookmarks. I am not sure why you also wrote »closed tabs«, since that seems to be a subset of history anyway.

In short I think technically this is definitely possible, however the more questionable issue is the problem of the UX. Currently the main and really **one and only** use case for this extension is to search for open tabs. It has been build this way and is streamlined to do this very well. More features can of course always be added, but I would find it hard to then still label this as 

> »Tab Ahead«
> Lightweight Chrome Extension that helps you to quickly find open tabs by title and url.

You could, of course, start to add this as a feature here, maybe adding a configuration to enable this as part of the settings-page. If we then decide it does not make sense to include it in this extension, we could also publish a totally new extension branded as a

> »Search for anything«
> Quickly find history items, bookmarks and open tabs

There are two points that speak against such an extension from my point of view:

1. Chrome's built-in omnibar already provides the functionality requested for history items and bookmarks.

2. And most importantly there are other extensions that already do **all of this**, like https://chrome.google.com/webstore/detail/quick-tabs/jnjfeinjfmenlddahdjdmgpbokiacbbb and https://chrome.google.com/webstore/detail/vimium/dbepggeogbaibhgnhhndojpepiihcmeb/

**Comment by mahdix** [comment]
I think `vimium` is providing much more feature than I am looking for, so I did not give it a try.

About `quick-tabs`, it seems to be what I want but I think it's not very strong when searching for history. I did the same search with it and also with another extension specifically for History and `quick-tabs` did not even manage to find it. 

Problem is there are good extensions which do a single search type well (open tabs or history or bookmarks), but cannot search other sources, but as a user, I want to see a webpage that I just remember it's title or part of url (Maybe it's in my history or bookmarks or an open tab). I am looking for a single extension which just searches for URLs based on my search criteria.
